### PR TITLE
Bootstrap repository with basic scaffold project

### PR DESCRIPTION
**Pull request by JoelSpeed**
This sets up the basics of the repository with the various hack scripts and makefile targets that we typically use in our Go projects. Add to that the very very basic manager configuration for the bootstrap and controlplane providers to prove the tooling is working.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,229 @@
+module github.com/openshift/cluster-api-provider-openshift
+
+go 1.20
+
+require (
+	github.com/golangci/golangci-lint v1.54.2
+	github.com/onsi/ginkgo/v2 v2.11.0
+	k8s.io/apimachinery v0.28.0
+	k8s.io/client-go v0.28.0
+	sigs.k8s.io/controller-runtime v0.16.0
```
Comment (damdo) [inline go.mod:10]: Looks like 0.16.2 is out

File: .golangci.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+linters:
```
Comment (damdo) [inline .golangci.yaml:1]: Is this the same config we have in CPMSO?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# cluster-api-provider-openshift
+
+This repository contains Cluster API provider implementations for the OpenShift container platform.
```
Comment (damdo) [inline README.md:3]: Is it worth mentioning we are implementing bootstrap and controlplane type of CAPI providers here?

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,229 @@
+module github.com/openshift/cluster-api-provider-openshift
+
+go 1.20
+
+require (
+	github.com/golangci/golangci-lint v1.54.2
+	github.com/onsi/ginkgo/v2 v2.11.0
+	k8s.io/apimachinery v0.28.0
+	k8s.io/client-go v0.28.0
```
Comment (damdo) [inline go.mod:9]: 0.28.2 is out

**Review by damdo — COMMENTED** [review]
/lgtm

File: .golangci.yaml (hunk 2 above)
Comment (JoelSpeed) [reply .golangci.yaml:1]: Yeah pretty much the same

File: README.md (hunk 3 above)
Comment (JoelSpeed) [reply README.md:3]: I think we can update this later, until we have something tangible I had planned to keep this relatively light

**Pull request merged**